### PR TITLE
refactor: use src layout and pyproject.toml

### PR DESCRIPTION
**Pull request by tlambert03**
see #280  for a little discussion.  setuptools editable installs now do a bad job unless you're using a src layout.  so this is necessary

### CODE REVIEW

**Comment by tlambert03** [comment]
the one broken napari test needs to be fixed on the napari side since #276 ... (and testing/fixing that was actually the thing that got me annoyed about setuptools in the first place)

**Comment by tlambert03** [comment]
(see https://github.com/napari/napari/pull/5698 for fix on napari side)

File: pyproject.toml [hunk 1]
```diff
@@ -3,12 +3,90 @@
...
+]
+classifiers = [
+    "Development Status :: 3 - Alpha",
+    "License :: OSI Approved :: BSD License",
+    "Natural Language :: English",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
Comment (Czaki) [inline pyproject.toml:26]: What is blocking us from adding support of python 3.11 for npe2 (I know problems with napari itself).

**Review by Czaki — APPROVED** [review]
Two minor comments

File: pyproject.toml (hunk 1 above)
Comment (tlambert03) [reply pyproject.toml:26]: Nothing, but yes, out of scope here

File: pyproject.toml [hunk 2]
```diff
@@ -3,12 +3,90 @@
...
+    "Typing :: Typed",
+]
+dependencies = [
+    "PyYAML",
+    "appdirs",
+    "build",
```
[the remark this answers is not in the document]
Comment (tlambert03) [reply pyproject.toml:32]: Perhaps, but can't remember, and out of scope here

File: pyproject.toml [hunk 3]
```diff
@@ -29,23 +107,63 @@ line-length = 88
...
+    "I",   # isort
+    "B",   # flake8-bugbear
+    "C4",  # flake8-comprehensions
+    "TID", # flake8-tidy-imports
+    "RUF", # ruff-specific rules
 ]
 
 [tool.ruff.per-file-ignores]
-"npe2/cli.py" = ["B008","A00"]
+"src/npe2/cli.py" = ["B008", "A00"]
```
Comment (nclack) [inline pyproject.toml:124]: Add comments for these?  Something like:

```python
# B008 - (flake8) Do not perform function call {name} in argument defaults
"src/npe2/cli.py" = ["B008", "A00"]
```

I tried to look up A00 but couldn't find it. Just want to double check you didn't mean something like A001.

**Review by nclack — APPROVED** [review]
lgtm!

Left one slightly out-of-scope comment. I had a question about the ruff ignore rules, but that predates this pr.

**Pull request merged**